user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -595,6 +610,66 @@ def _read_qualification_output_file(self, report_name_key: str, file_format_key:
         report_file_path = FSUtil.build_path(self.ctxt.get_rapids_output_folder(), report_file_name)
         return pd.read_csv(report_file_path)
 
+    def _read_qualification_metric_file(self, file_name: str) -> Dict[str, pd.DataFrame]:
+        """
+        Helper method to read metric files from the qualification tool's output metric folder.
+        Returns a dictionary of DataFrames, where each key is an application ID, and each
+        DataFrame contains the corresponding application's metrics data.
+        Example:
+            {
+                'appId1': pd.DataFrame(...),
+                'appId2': pd.DataFrame(...),
+            }
+       :param file_name: Name of the metric file to read from each application's folder
+       """
+        metrics = {}
+        root_metric_dir = self.ctxt.get_metrics_output_folder()
+        for metric_dir in FSUtil.get_subdirectories(root_metric_dir):
+            app_id_str = FSUtil.get_resource_name(metric_dir)
+            report_file_path = FSUtil.build_path(metric_dir, file_name)
+            try:
+                metrics[app_id_str] = pd.read_csv(report_file_path)
+            except Exception as e:  # pylint: disable=broad-except
+                # Some apps may not have the given metrics file, we should ensure
+                # that the dictionary contains entries for all apps to avoid KeyErrors
+                # and maintain consistency in processing.
+                metrics[app_id_str] = pd.DataFrame()
+                self.logger.warning('Unable to read metrics file for app %s. Reason - %s:%s',
+                                    app_id_str, type(e).__name__, e)
+        return metrics
+
+    def _assign_execution_engine_to_apps(self, tools_processed_apps: pd.DataFrame) -> pd.DataFrame:
+        """
+        Assigns the execution engine (Spark/Photon) to each application. This will be used to categorize
+        applications into speedup categories (Small/Medium/Large).
+        """
+        spark_properties = self._read_qualification_metric_file('spark_properties.csv')
+        default_exec_engine_type = ExecutionEngine.get_default()
+        exec_engine_col_name = self.ctxt.get_value('local', 'output', 'speedupCategories', 'execEngineColumnName')
+
+        # Default to Spark-based execution type for non-Databricks platforms
+        if self.ctxt.platform.get_platform_name() not in {CspEnv.DATABRICKS_AWS, CspEnv.DATABRICKS_AZURE}:
+            tools_processed_apps[exec_engine_col_name] = default_exec_engine_type
+            return tools_processed_apps
+
+        # Create a map of App IDs to their execution engine type (Spark/Photon)
+        spark_version_key = 'spark.databricks.clusterUsageTags.sparkVersion'
```

Comment:
same here..it is a platform specific that should not be defined by the Qual tool.

Reply:
This comment may not be relevant as the function has been simplified. The logic to parse the runtime has been moved to Scala code.